include/ICPUBuffer.h

```diff
@@ -37,10 +37,18 @@ class ICPUBuffer : public IBuffer
             size = sizeInBytes;
         }
 
+		//! Returns buffer's type which in this case is always equal to `E_BUFFER_TYPE::EBT_UNSPECIFIED_BUFFER`.
```

Comment:
we initially wanted to support buffer typing (like in DX10) but OpenGL  4.0+ brought ARB_buffer_storage which makes the buffer type irrelevant, since you explicitly state if the buffer should be on the GPU or not.
Aside from that OpenGL never had strict buffer typing, the buffer type was more of a hint as to whether to store the buffer in RAM,DMA RAM or VRAM.
So this E_BUFFER_TYPE enum is deprecated.